src/ios/ZeroConf.swift

```diff
@@ -436,4 +433,36 @@ import Foundation
         return nil
     }
 
+    fileprivate static func dictionary(fromTXTRecord txtData: Data) -> [String: String] {
+
+        var result = [String: String]()
+        var data = txtData
+
+        while !data.isEmpty {
+            // The first byte of each record is its length, so prefix that much data
+            let recordLength = Int(data.removeFirst())
+            guard data.count >= recordLength else { return [:] }
+            let recordData = data[..<(data.startIndex + recordLength)]
+            data = data.dropFirst(recordLength)
+
+            guard let record = String(bytes: recordData, encoding: .utf8) else { return [:] }
```

Comment:
It looks like we now have 3 possible return values from this method:

1. Empty key-value (`[:]`)
1. Populated key, but empty value (`case 1: result[key] = ""`)
1. Populated key and value (`case 2: result[key] = String(keyValue[1])`)

Previously I think if there were no records at all then the final JSON response looks like `"txtRecord": {}`. What does it look like now in the event that one of these guards returns `[:]`?

These guards might be a good spot to do some debug printing too.